09_maps/main.go

```diff
@@ -111,18 +134,268 @@ func main() {
-	// prints the average BMI of the people in the array.
+	// calculate average BMI
+	fmt.Println("Avg BMI is", calcAvgBMI(people))
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// returns the map with the highest BMI.
+	// calculate highest BMI
+	fmt.Println("Highest BMI is", highestBMI(people))
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// add as a new key-value pair to each map the key "bmi" with the value
-	// being the BMI of the corresponding person.
-	// Hint: Use a loop.
-	// Hint: Create a new map for each person.
-	// Hint: Create a new array of maps.
+	// add BMI to array of maps
+	fmt.Println(addBMI(people))
+
+	// test helper int function
+	fmt.Println(largestInt([]int{}))                      // 0
+	fmt.Println(largestInt([]int{1, 2, 3}))               // 3
+	fmt.Println(largestInt([]int{-1, -2, 0}))             // 0
+	fmt.Println(largestInt([]int{100, 0, 3000, 0, -100})) // 3000
+
+	// test helper string function
+	fmt.Println(largestString([]string{"", ""}))           // ""
+	fmt.Println(largestString([]string{"a", "bc", "def"})) // "def"
+
+	// Close main function, to be able to declare another function
+}
+
+// create empty map
+// create function
+// create counter
+// create loop
+// create break condition if counter is > than the key-value pairs the map has
+// operate on the input value of the keys and do math on it
+// add the result to a new map
+// add a new key value element and fill it depending on the value of the input height key
+// print new map
+
+func createNewMap(mapInput map[string]int) {
+	var newMap map[string]int = map[string]int{}
+	var counter int = 0
+
+	for {
+		if counter > len(mapInput) {
+			break
+		}
+		newMap["age"] = mapInput["age"] + 2
+		newMap["height"] = mapInput["height"] * 2
+		newMap["weight"] = mapInput["weight"] * 2
+		if mapInput["height"] > 200 {
+			newMap["tall"] = 1
+		}
+		if mapInput["height"] <= 200 {
+			newMap["tall"] = 0
+		}
+		counter++
+	}
+	fmt.Println("createNewMap:", newMap)
+}
+
+// TODO: Create a function that takes a map as an argument and prints the
+// BMI of the person. The formula for BMI is:
+// BMI = weight / (height * height)
+// If the map does not have the key "height", print "Height not found".
+// If the map does not have the key "weight", print "Weight not found".
+// Otherwise, print "<name>'s BMI is <bmi>".
+// Hint: Use this function in later exercises.
+
+func calcBMI(mapInput map[string]int) float64 {
+	var value int
+	var exists bool
+	var bmi float64
+	var heightInt int = mapInput["height"]
+	var weightInt int = mapInput["weight"]
+
+	value, exists = mapInput["height"]
+	if !exists || value == 0 {
+		fmt.Println("Height not found")
+	}
+	var heightFloat float64 = float64(heightInt)
+	value, exists = mapInput["weight"]
+	if !exists || value == 0 {
+		fmt.Println("Weight not found")
+	}
+	var weightFloat float64 = float64(weightInt)
```

Comment:
You are getting `mapInput["height"]` and `mapInput["weight"]` twice, this is unnecessary.

Can you find a way to simplify this further?

Reply:
I commented out both variables, as they are just used to immediately be use for the float64 variables. Not sure you meant that?